src/main/java/org/fogbowcloud/arrebol/execution/DockerTaskExecutor.java

```diff
@@ -1,146 +1,200 @@
     @Override
-    public TaskExecutionResult execute(Task task) {
-
+    public TaskExecutionResult execute(Task task){
         //FIXME: We should catch the errors when starting/finishing the container and move the task to its FAILURE state
         //FIXME: also, follow the SAME log format we used in the RawTaskExecutor
-
         TaskExecutionResult taskExecutionResult;
+
         Integer startStatus = this.start();
 
+        updateRequirements(task.getTaskSpec());
+
         if(startStatus != 0){
             LOGGER.error("Exit code from container start: " + startStatus);
-            throw new DockerStartException("Could not start container " + this.containerName);
+            throw new DockerStartException("Could not start container " + getContainerName());
         }
 
-        LOGGER.info("Successful started container " + this.containerName);
+        LOGGER.info("Successful started container " + getContainerName());
+        Command[] commands = getCommands(task);
+        LOGGER.info("Starting to execute commands [" + commands.length + "] of task " + task.getId());
+        int[] commandsResults = executeCommands(commands);
+
+        Integer stopStatus = this.stop();
+        if(stopStatus != 0){
+            LOGGER.error("Exit code from container stop: " + stopStatus);
+        }
 
+        taskExecutionResult = getTaskResult(commands, commandsResults);
+
+        LOGGER.info("Result of task [" + task.getId() + "]: " + taskExecutionResult.getResult().toString());
+        return taskExecutionResult;
+    }
+
+    private void updateRequirements(TaskSpec taskSpec){
+        verifyImage(taskSpec);
+        Map<String, String> mapRequirements = new HashMap<>();
+        String dockerRequirements = taskSpec.getSpec().getRequirements().get(DockerConstants.METADATA_DOCKER_REQUIREMENTS);
+        if(dockerRequirements != null){
+            String requirements[] = dockerRequirements.split("&&");
+            for(String requirement : requirements){
+                String req[] = requirement.split("==");
+                String key = req[0].trim();
+                String value = req[1].trim();
+                switch (key){
+                    case DockerConstants.DOCKER_MEMORY:
+                        mapRequirements.put(DockerConstants.JSON_KEY_MEMORY, value);
+                        LOGGER.info("Added requirement ["+DockerConstants.JSON_KEY_MEMORY+"] with value ["+value+"]");
+                        break;
+                    case DockerConstants.DOCKER_CPU_WEIGHT:
+                        mapRequirements.put(DockerConstants.JSON_KEY_CPU_SHARES, value);
+                        LOGGER.info("Added requirement ["+DockerConstants.JSON_KEY_CPU_SHARES+"] with value ["+value+"]");
+                        break;
+                }
+            }
+            this.workerDockerRequestHelper.setRequirements(mapRequirements);
+        }
+    }
+
+    private void verifyImage(TaskSpec taskSpec){
+        String image = taskSpec.getImage();
```

Comment:
why a method named "verify" changes the inner state of an object? we have two problems here: i) a naming problema; and 2) a design problem (I'd avoid this object to have mutable state)